src/main/java/org/opensearch/ad/transport/ValidateAnomalyDetectorRequest.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.ad.transport;
+
+import java.io.IOException;
+
+import org.opensearch.action.ActionRequest;
+import org.opensearch.action.ActionRequestValidationException;
+import org.opensearch.ad.model.AnomalyDetector;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.unit.TimeValue;
+
+public class ValidateAnomalyDetectorRequest extends ActionRequest {
+
+    private final AnomalyDetector detector;
+    private final String typeStr;
```

Comment:
`typeStr` means the validation type, `detector` or `model`?

Reply:
Yes, will change to validationType